unity-renderer/Assets/Scripts/MainScripts/DCL/Components/AvatarAttach/AvatarAttachHandler.cs

```diff
@@ -0,0 +1,121 @@
+using System.Collections;
+using DCL.Configuration;
+using DCL.Controllers;
+using DCL.Models;
+using UnityEngine;
+
+namespace DCL.Components
+{
+    internal class AvatarAttachHandler
+    {
+        public AvatarAttachComponent.Model model { internal set; get; } = new AvatarAttachComponent.Model();
+        public IParcelScene scene { private set; get; }
+        public IDCLEntity entity { private set; get; }
+
+        private AvatarAttachComponent.Model prevModel = null;
+
+        private IAvatarAnchorPoints anchorPoints;
+        private AvatarAnchorPointIds anchorPointId;
+
+        private Coroutine componentUpdate = null;
+
+        private readonly AvatarAttachPlayerHandler avatarAttachPlayerHandler = new AvatarAttachPlayerHandler();
+        private ISceneBoundsChecker sceneBoundsChecker => Environment.i?.world?.sceneBoundsChecker;
+
+        public void Initialize(IParcelScene scene, IDCLEntity entity)
+        {
+            this.scene = scene;
+            this.entity = entity;
+            avatarAttachPlayerHandler.onAvatarDisconnect += Detach;
+        }
+
+        public void OnModelUpdated(string json)
+        {
+            OnModelUpdated(model.GetDataFromJSON(json) as AvatarAttachComponent.Model);
+        }
+
+        public void OnModelUpdated(AvatarAttachComponent.Model newModel)
+        {
+            prevModel = model;
+            model = newModel;
+
+            if (model == null)
+            {
+                return;
+            }
+
+            sceneBoundsChecker?.AddEntityToBeChecked(entity);
```

Comment:
This is not needed. The `sceneBoundsChecker` maintenance is already taken care of by `ParcelScene`.